src/pcap/bosh.go

```diff
@@ -1,55 +1,553 @@
 		return fmt.Errorf("invalid message: token: %w", errEmptyField)
 	}
 
-	if request.Deployment == "" {
+	if boshRequest.Deployment == "" {
 		return fmt.Errorf("invalid message: deployment: %w", errEmptyField)
 	}
 
-	if len(request.Groups) == 0 {
+	if boshRequest.Environment == "" {
+		return fmt.Errorf("invalid message: environment: %w", errEmptyField)
+	}
+
+	if len(boshRequest.Groups) == 0 {
 		return fmt.Errorf("invalid message: instance group(s): %w", errEmptyField)
 	}
 
 	return nil
 }
+
+// setup is called in NewBoshResolver and ensures that a connection to the configured BOSH director is possible.
+// Retrieves information about the director and its UAA.
+//
+// Returns an error if the connection to the BOSH Director is not possible.
+func (br *BoshResolver) setup() error {
+	br.logger.Debug("setting up BoshResolver", zap.Any("resolver-config", br.Config))
+
+	var tlsConfig *tls.Config
+
+	if br.Config.MTLS != nil {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+			MaxVersion: tls.VersionTLS13,
+			ClientAuth: tls.RequireAndVerifyClientCert,
+			RootCAs:    br.boshRootCAs,
+		}
+	}
+
+	timeout := 500 * time.Millisecond //nolint:gomnd // reasonable value.
+
+	br.client = &http.Client{
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: timeout,
+			}).DialContext,
+			TLSHandshakeTimeout:   timeout,
+			ResponseHeaderTimeout: timeout,
+			ExpectContinueTimeout: timeout,
+			DisableKeepAlives:     true,
+			MaxIdleConnsPerHost:   -1,
+			TLSClientConfig:       tlsConfig,
+		},
+		Timeout: timeout,
+	}
+
+	br.logger.Debug("discovering bosh-UAA endpoint", zap.String("bosh-director", br.DirectorURL.String()))
+	apiResponse, err := br.info()
+	if err != nil {
+		return err
+	}
+
+	br.UaaURLs = apiResponse.UserAuthentication.Options.URLs
+	br.logger.Info("connected to bosh-director", zap.Any("bosh-director", br.DirectorURL.String()))
+	return nil
+}
+
+// info retrieves the BOSH director /info endpoint.
+//
+// Used for startup and health check.
+//
+// Returns an error if the connection to the director URL fails or the response from the director is not a BoshInfo.
+func (br *BoshResolver) info() (*BoshInfo, error) {
+	infoEndpoint := br.DirectorURL.JoinPath("/info")
+
+	response, err := br.client.Do(&http.Request{
+		Method: http.MethodGet,
+		URL:    infoEndpoint,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch Bosh Director API from %v: %w", br.Config.RawDirectorURL, err)
+	}
+
+	defer func() { _ = response.Body.Close() }()
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("received non-OK response from Bosh Director: %s", response.Status)
+	}
+
+	var apiResponse *BoshInfo
+	data, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("could not read bosh-director API response: %w", err)
+	}
+	err = json.Unmarshal(data, &apiResponse)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse bosh-director API response: %w", err)
+	}
```

Comment:
Please use `json.NewDecoder(response.Body).Decode(&apiResponse)`.

But this shouldn't work anyway because the apiResponse pointer is nil since it has never been initialized and you are taking a pointer to a pointer.